azerothcore/azerothcore-wotlk
# refactor[Core/reputation_spillover_template]: Reputation spillover table expansion

## Changes Proposed:
-  Update to `ObjectMgr::LoadReputationSpilloverTemplate()`, allowing two more factions to be added to `reputation_spillover_template` table. Allows more race-specific factions to have correct reputation gains from questing (like, for example Gilneas/Bilgewater Cartel that could be added in https://github.com/heyitsbench/mod-worgoblin without needs to do custom changes to ObjectMgr.cpp).

## Issues Addressed:
- None I know about.

## Tests Performed:
- Tested on Win10Home(22H2) but I assume it will work on any OS, it basically just builds upon pre-existing function. Tests done during playtesting of Faction.dbc bugfixes I've commited to Worgoblin module with backported Gilneas and Bilgewater Cartel from Cataclysm. As far as I can tell, everything was behaving correctly.


## How to Test the Changes:
1. Create new entry in `Factions.dbc` that has either alliance or horde as a parent faction (could just duplicate Stormwind/Orgrimmar with changed ID [and name to avoid confusion during playtesting]), then add said entry's ID to `reputation_spillover_template` in `acore_world` database under `faction5` (also fill `rate_5` and `rank_5` respectively): for alliance, for factions 47, 54, 69, 72 and 930, for horde factions 68, 76, 81, 530 and 911. Not 100% sure but `Factions.dbc` should probably be identical between server and client.
2. Create new character of any race that belongs to side you created new faction for and enter the world.
3. Do a couple of starting zone quests observing your reputation tab, if everything is done correctly, character should receive a fraction of quest reputation reward for every turn in. 

## Known Issues and TODO List:
- Limits are still in place, if someone wanted to add more than 6 reputations, would still need to update core. I've made this PR mosty with Worgoblin module in mind, leaving a bit of leeway for additional race(s) (so, for example pandaren in the future).

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
